src/v/kafka/server/request_context.h

```diff
@@ -248,12 +254,71 @@ class request_context {
         return _conn->server().credentials();
     }
 
+    // Set force to true to always enqueue the audit event regardless of event
+    // type
+    template<typename T = model::topic>
+    bool audit(
+      security::audit::event_type event_type,
+      std::vector<T> additional_resources = {}) {
+        auto operation_name = handler_for_key(_header.key).value()->name();
+        auto& audit_mgr = _conn->server().audit_mgr();
+        auto additional_resource_details
+          = security::audit::create_resource_details(additional_resources);
+        _authz_results.erase(
+          std::remove_if(
+            _authz_results.begin(),
+            _authz_results.end(),
+            [this,
+             operation_name,
```

Comment:
std::move

Reply:
actually `operation_name` is a `const char *` so move doesn't do anything.